base/sort.jl

```diff
@@ -85,9 +85,16 @@ julia> issorted([1, 2, -2, 3], by=abs)
 true
 ```
 """
-issorted(itr;
-    lt=isless, by=identity, rev::Union{Bool,Nothing}=nothing, order::Ordering=Forward) =
-    issorted(itr, ord(lt,by,rev,order))
+function issorted(itr;
+        lt=isless, by=identity, rev::Union{Bool,Nothing}=nothing, order::Ordering=Forward)
+    # Explicit branching because the compiler can't optimize away the
+    # type instability of the `ord` call with Bool `rev` parameter.
+    if rev === true
+        issorted(itr, ReverseOrdering(ord(lt,by,nothing,order)))
```

Comment:
`ord(lt,by,nothing,order)` is duplicated. Could you instead assign it to a variable before the `if` statement, and then use the variable in both branches?

Reply:
As I was making that change I grepped for `=== false` and found this paradigm repeated many times through the file in various formats, so I copied the style used elsewhere which does contain two `ord` calls but does not have redundancy.